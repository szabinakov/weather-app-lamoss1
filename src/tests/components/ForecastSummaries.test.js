import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1525046400000,
        temperature: {
          max: 11,
          min: 4,
        },
        description: "Clear",
        icon: 800,
      },
      {
        date: 1525132800000,
        temperature: {
          max: 13,
          min: 8,
        },
        description: "Stormy",
        icon: 211,
      },
      {
        date: 1525219200000,
        temperature: {
          max: 1,
          min: -2,
        },
        description: "Heavy Snow",
        icon: 602,
      },
      {
        date: 1525305600000,
        temperature: {
          max: 20,
          min: 4,
        },
        description: "Tornado",
        icon: 781,
      },
      {
        date: 1525392000000,
        temperature: {
          max: 25,
          min: 18,
        },
        description: "Hazy",
        icon: 721,
      },
    ],
    onForecastSelect: () => {},
  };

  describe("ForecastSummaries", () => {
    it("renders correctly", () => {
      const { asFragment } = render(
        <ForecastSummaries
          forecasts={validProps.forecasts}
          onForecastSelect={validProps.onForecastSelect}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("renders the correct number of ForecastSummary instances", () => {
      const { getAllByTestId } = render(
        <ForecastSummaries forecasts={validProps.forecasts} />
      );

      expect(getAllByTestId("forecast-summary")).toHaveLength(5);
    });
  });
});
