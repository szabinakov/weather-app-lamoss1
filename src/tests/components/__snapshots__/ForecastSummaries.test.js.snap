// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1525046400000
      </div>
      Mon Apr 30 2018
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="800"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        11
      </div>
      <div
        class="forecast-summary__description"
      >
        Clear
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1525132800000
      </div>
      Tue May 01 2018
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="211"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        13
      </div>
      <div
        class="forecast-summary__description"
      >
        Stormy
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1525219200000
      </div>
      Wed May 02 2018
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="602"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        1
      </div>
      <div
        class="forecast-summary__description"
      >
        Heavy Snow
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1525305600000
      </div>
      Thu May 03 2018
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="781"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        20
      </div>
      <div
        class="forecast-summary__description"
      >
        Tornado
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1525392000000
      </div>
      Fri May 04 2018
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="721"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        25
      </div>
      <div
        class="forecast-summary__description"
      >
        Hazy
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
